### PR TITLE
Introduce internal plugin

### PR DESCRIPTION
**Pull request by oldergod**
Centralizing configurations. Not able to get the extension to work with included builds, no idea why.

### CODE REVIEW

File: build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    val spotless = extensions.getByName("spotless") as SpotlessExtension
+    val licenseHeaderFile = rootProject.file("gradle/license-header.txt")
+    spotless.apply {
+      // The nested build-support Gradle project contains Java sources. Use our root project to
+      // target its sources rather than duplicating the Spotless setup in multiple places.
+      java {
+        if (path == ":") {
+          target("build-support/settings/src/**/*.java")
+        } else {
+          target("src/**/*.java")
```
Comment (JakeWharton) [inline build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt:123]: Looks like you need an exclude here for test fixtures that have nested gradle builds

**Review by JakeWharton — APPROVED** [review]
Looks like a pretty good start

File: build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt (hunk 1 above)
Comment (oldergod) [reply build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt:123]: Is it what's creating the task dependency problems with spotless?

Comment (oldergod) [reply build-support/src/main/kotlin/com/squareup/wire/buildsupport/WireBuildPlugin.kt:123]: Got it, I saw the CI failures

**Pull request merged**